wrangler-transform/src/main/java/co/cask/wrangler/Wrangler.java

```diff
@@ -233,6 +239,38 @@ public void configurePipeline(PipelineConfigurer configurer) throws IllegalArgum
     }
   }
 
+  @Override
+  public void prepareRun(StageSubmitterContext context) throws Exception {
+    super.prepareRun(context);
+    List<String> inputFields = new ArrayList<>();
+    List<String> outputFields = new ArrayList<>();
+    Schema inputSchema = context.getInputSchema();
+    if (checkSchema(inputSchema, "input")) {
+      //noinspection ConstantConditions
+      inputFields = inputSchema.getFields().stream().map(Schema.Field::getName).collect(Collectors.toList());
+    }
+    Schema outputSchema = context.getOutputSchema();
+    if (checkSchema(outputSchema, "output")) {
+      //noinspection ConstantConditions
+      outputFields = outputSchema.getFields().stream().map(Schema.Field::getName).collect(Collectors.toList());
+    }
+    FieldOperation dataPrepOperation = new FieldTransformOperation("DataPrep", config.directives, inputFields,
+                                                                   outputFields);
+    context.record(Collections.singletonList(dataPrepOperation));
+  }
+
+  private boolean checkSchema(Schema schema, String name) {
+    if (schema == null) {
+      LOG.debug(String.format("The %s schema is null.", name));
```

Comment:
Log messages should tell somebody reading it everything they need to know without looking at any context. Better to make it:

is null. Field level lineage will not be recorded.